src/mlpack/methods/reinforcement_learning/environment/bit_flipping.hpp

```diff
@@ -0,0 +1,256 @@
+/**
+ * @file methods/reinforcement_learning/environment/bit_flipping.hpp
+ * @author Eshaan Agarwal
+ *
+ * This file is an implementation of Bit Flipping toy task:
+ * https://github.com/NervanaSystems/gym-bit-flip/blob/master/gym_bit_flip/bit_flip.py
+ * https://github.com/ceteke/her/blob/master/env.py
+ *
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_RL_ENVIRONMENT_BIT_FLIPPING_HPP
+#define MLPACK_METHODS_RL_ENVIRONMENT_BIT_FLIPPING_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+
+/**
+ * Implementation of Cart Pole task.
+ */
+class BitFlipping
+{
+ public:
+  /**
+   * Implementation of the state of Bit Flipping. 
+   * Each state is a n length binary vector
+   */
+  class State
+  {
+   public:
+    /**
+     * Construct a state instance.
+     */
+    State() : data(dimension)
+    { /* Nothing to do here. */ }
+
+    /**
+     * Construct a state instance from given data.
+     *
+     * @param data Data 
```

Comment:
Missing description.

Reply:
I actually copied this from other environments. I will write some for decsription then.